src/components/Note.tsx

```diff
@@ -27,7 +27,15 @@ const Note = ({ context, onFocus }: NoteProps) => {
   const [justPasted, setJustPasted] = useState(false)
 
   const hasNote = hasChild(state, context, '=note')
-  if (!hasNote || isContextViewActive(state, context)) return null
+
+  /** Check if the note thought is pending or not. */
+  const isNotePending = () => {
+    const hashedContext = hashContext([...context, '=note'])
+    const noteThought = state.thoughts.contextIndex[hashedContext]
```

Comment:
Avoid accessing `state.thoughts` directly. That is a low-level API that should only be accessed in low-level thought edit operations.

You can use the isPending selector: 

https://github.com/cybersemics/em/blob/b03a97a8b1a7cc0c53ce253b0ef9a16f33429510/src/selectors/isPending.ts#L5-L6